tests/extractcode/test_extract.py

```diff
@@ -1112,9 +1112,24 @@ def test_extract_zipslip_tar_posix(self):
 
         warns = [r.warnings for r in result if r.warnings]
         assert [] == warns
-
+        
     def test_extract_always_returns_a_generator_and_not_a_list(self):
         # a test for #1996 to ensure that progress is displayed "progressively"
         test_dir = self.get_test_loc('extract/generator', copy=True)
         result = extract.extract(test_dir)
         assert isinstance(result, GeneratorType)
+    
+    def test_extract_ignore(self):
```

Comment:
We probably want to have more test cases here. We want to make sure `--ignore` can ignore:
1) full filenames
2) directory names
3) various patterns

We should at least have tests cases for all of the above. I would review the current set of extractcode tests and model them after that.